File: leetcode-python-go/1710. 卡车上的最大单元数/main.py
from typing import List

boxTypes = [[1,3],[2,2],[3,1]]
truckSize = 4

def maximumUnits(boxTypes: List[List[int]], truckSize: int) -> int:
    boxTypes.sort(key=lambda x: x[1], reverse=True)
    res = 0
    for numberOfBoxes, numberOfUnitsPerBox in boxTypes:
        if numberOfBoxes >= truckSize:
            res += truckSize * numberOfUnitsPerBox
            break
        res += numberOfBoxes * numberOfUnitsPerBox
        truckSize -= numberOfBoxes
    return res


print(maximumUnits(boxTypes, truckSize))

File: leetcode-python-go/1710. 卡车上的最大单元数/main.go
package main

import (
	"fmt"
	"sort"
)

func maximumUnits(boxTypes [][]int, truckSize int) (ans int) {
	sort.Slice(boxTypes, func(i, j int) bool { return boxTypes[i][1] > boxTypes[j][1] })
	for _, p := range boxTypes {
		if p[0] >= truckSize {
			ans += truckSize * p[1]
			break
		}
		truckSize -= p[0]
		ans += p[0] * p[1]
	}
	return
}

func main() {
	truckSize := 4
	boxTypes := [][]int{{1, 3}, {2, 2}, {3, 1}}
	results := maximumUnits(boxTypes, truckSize)
	fmt.Println(results)
}
